File: baekjoon/exhaustiveSearch/1476.py
import sys

input = sys.stdin.readline


E, S , M = map(int,input().split())

E2 = S2 = M2 = 1
years = 1
while True:
    if E2 == E and S == S2 and M == M2:
        break

    if E2 < 15:
        E2 += 1
    else:
        E2 = 1

    if S2 < 28:
        S2 += 1
    else:
        S2 = 1

    if M2 < 19:
        M2 += 1
    else:
        M2 = 1

    years += 1

print(years)

File: baekjoon_js/exhaustiveSearch/1476.js
("use strict");

const fs = require("fs");

const stdin = (
  process.platform === "linux"
    ? fs.readFileSync("/dev/stdin").toString()
    : `15 28 19
`
).split("\n");

const input = (() => {
  let line = 0;
  return () => stdin[line++];
})();

const [E, S, M] = input().split(" ").map(Number);

let days = 1;
let tempE = 1;
let tempS = 1;
let tempM = 1;
// console.log(tempE, tempS, tempM);
while (true) {
  if (tempE === E && tempS === S && tempM == M) break;
  days++;
  if (tempE + 1 < 16) tempE++;
  else tempE = 1;
  if (tempS + 1 < 29) tempS++;
  else tempS = 1;
  if (tempM + 1 < 20) tempM++;
  else tempM = 1;
}

console.log(days);
